reactor-test/build.gradle

```diff
@@ -40,10 +40,11 @@ dependencies {
 
 	optional "org.jetbrains.kotlin:kotlin-stdlib:${kotlinVersion}"
 
-	testCompile "junit:junit:$jUnitVersion"
-	testImplementation "org.junit.jupiter:junit-jupiter-api:${jUnitJupiterVersion}"
-	testRuntime "org.junit.jupiter:junit-jupiter-engine:${jUnitJupiterVersion}"
-	testRuntime "org.junit.vintage:junit-vintage-engine:${jUnitJupiterVersion}"
+	implementation platform("org.junit:junit-bom:${jUnitPlatformVersion}")
```

Comment:
same as above

Reply:
it should actually go to the top level `build.gradle` for every subproject

Reply:
> it should actually go to the top level build.gradle for every subproject

Yeah, tried something similar at first but didn't work and was worried that it wouldn't be clear where the magic came from.

Reply:
Since I have a strong preference to declare the the version strings in the root/`gradle.properties` and the dependencies in each module's `build.gradle` file, to me it makes sense to colocate the declaration of the platform in the module build files (even if it means duplicating a bit). But either way would be fine I think.

Reply:
> Yeah, tried something similar at first but didn't work and was worried that it wouldn't be clear where the magic came from.

could you please try again and, if it does not work, I am happy to help making it working :) 

> Since I have a strong preference to declare the the version strings in the root/gradle.properties and the dependencies in each module's build.gradle file,

That's how it will be: the platform (aka "the dependency management") in the root file, and dependencies (like `junit-platform-api`) in projects' files.

Reply:
That's debatable. I'd like one central place to define the _versions_ so that bumping a version impacts all modules that define a dependency on that artifact. But I want the actual declaring of the dependency done in each module build file. The `platform` instruction is in a bit of a gray area I'd say. What about modules that don't JUnit5 ? In fact, we have `benchmarks` which doesn't, so it tends to make me think the `platform` should be treated like the actual dependencies.

Reply:
> The platform instruction is in a bit of a gray area I'd say. What about modules that don't JUnit5 ? In fact, we have benchmarks which doesn't, so it tends to make me think the platform should be treated like the actual dependencies.

adding platform *does not* add junit as a dependency.

Reply:
I'm actually more in synch with @simonbasle

Reply:
I know, **but it doesn't make sense to apply a platform to a module if none of the dependencies it curates are going to be used by that module.** So better put it next to the dependencies themselves rather than as a blanket instruction via `configure(subprojects)` in the root build file (which also has the downside of adding a third place to look at to get the whole dependency picture).

Having a split between `xxxVersion` variable and dependency declarations is acceptable because it brings the benefit of facilitated version bumps.

Reply:
Guys, if you want "all dependency management to be in the root `build.gradle` and dependencies defined in per-project files", than you really should put `platform` to the root `build.gradle`, because _that's how you apply the dependency management in Gradle_.

You can be in sync with each other, but it may also mean that you may both misunderstood Gradle's (overly complicated) dependency management. But that's how it works. Adding `platform(...)` "dependency" does not make the module _depend_ on said project.

Reply:
@simonbasle @ericbottard folks, just sharing what we do at rsocket-java - 

we define `dependencyManagement` section for all the projects once at the root -> https://github.com/rsocket/rsocket-java/blob/master/build.gradle#L56 and then all subproject uses `dependencies` with no versions which make it much cleaner. -> https://github.com/rsocket/rsocket-java/blob/master/rsocket-core/build.gradle#L26.

I believe that does not contradict to "central place for all the versions" but rather enforce it so just in case somebody put a version is sub `build.gradle` this will be more than noticeable

Reply:
> Having a split between xxxVersion variable and dependency declarations is acceptable because it brings the benefit of facilitated version bumps.

if we have a single `platform()` entry in the whole build than we woudn't even need variables on a first place. But okay, I see that you got confused with Gradle's dependency management _syntax_ (where you need to depend on a platform although you're not depending on anything), so I refrain from this debase, especially given that Eric is more in sync with you.

Reply:
> if we have a single platform() entry in the whole build than we woudn't even need variables on a first place.

ah but you're talking about _defining_ an ad-hoc platform? Vs _importing_ a "platform" (in the "an existing Maven BOM" sense)... which Gradle documents as a **regular dependency declaration** in their docs:

> The BOM support in Gradle works similar to using <scope>import</scope> when depending on a BOM in Maven. In Gradle however, it is done via a regular dependency declaration on the BOM:

Unfortunately Gradle only supports defining a platform in a module that cannot be also a `java` or `java-library` module, ie. one that is intended to be published). So no adhoc platform definition, and we're stuck with variables for now.

Reply:
we could maybe switch to the `dependencyManagement` model with the spring plugin

Reply:
I think we do understand how gradle works, in particular the fact that depending on a `platform` does not actually add dependencies. This is all about gathering of information in one or several places. I'm personally okay with both, as I see arguments for both, even though I prefer Simon's view that "if we depend on junit platform in project X, then we see (all of) that in that project (the bom and the actual dependencies)".

As a matter of fact, there may be a case in the future where we (need to) depend on a given specific version of junit-bom in one project and are fine with another in the other projects. So while the desire to rationalize at top level is appealing, we shouldn't force it too much IMO.

So to recap, I suggest to move the `platform` bit back to each project (while still using a version coming from a top level property for now). Is that ok ?